src/python/pants/engine/desktop.py

```diff
@@ -1,83 +1,67 @@
 # Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
```

Comment:
Same changes as the original implementation so that opening a file uses the engine correctly.

Note that the `InteractiveProcess` will be hermetic. We use `BinaryPaths` to find the `open` program, so there is no reason to have a non-hermetic env.